content/post/why-upstreaming-matters/index.md

```diff
@@ -0,0 +1,33 @@
+---
+title: Why upsteaming patches matters
+description: The benefits and reasons why you should always upstream patches.
+date: 2024-06-10 18:18:00 +0800
+categories:
+    - nix
+---
+
+With packaging software, there will always be cases where software fail to run
+or build as expected. This is especially common in Nix where it is not a traditional
+Linux/UNIX environment. For example, Nix does not use the filesystem hierarchy (FHS)
+which is standard in POSIX. Instead each package has its own directory in the Nix
+store, also referred to as a local cache, and contains directories such as `bin`,
+`lib`, `share`, etc. As most software is written to expect the FHS, they can easily
+break and not run as expected. The Nix store is also writable at build time of a
+package but ready only after that. This means the Nix store is essentially immutable.
```

Comment:
Alternative phrasing: `The Nix store is only writable at build time, becoming essentially immutable thereafter.`

Reply:
Alright, rephrased it.